Production/python/customiseHLT.py

```diff
@@ -0,0 +1,159 @@
+    extension = cms.bool(False), # this is the main table
+    variables = cms.PSet(
+      P4Vars,
+      charge = Var("charge", int, doc="electric charge"),
+    )
+  )
+  # cms.EDProducer : an object that produces a new data object
+
+  process.pfCandTable = cms.EDProducer("SimpleCandidateFlatTableProducer",
+    src = cms.InputTag( "hltParticleFlowForTaus" ),
+    cut = cms.string(""),
+    name= cms.string("PFCand"),
+    doc = cms.string("HLT PF candidates for taus"),
+    singleton = cms.bool(False), # the number of entries is variable
+    extension = cms.bool(False), # this is the main table
+    variables = cms.PSet(
+      P4Vars,
+      charge = Var("charge", int, doc="electric charge"),
+
+      energy = Var("energy",float, doc= 'energy'),
+      et = Var("et",float, doc= 'transverse energy'),
+      et2 = Var("et2",float, doc= 'transverse energy squared (use this for cut!)'),
+      massSqr = Var("massSqr",float, doc= 'mass squared'),
+      mt = Var("mt",float, doc= 'transverse mass'),
+      mtSqr = Var("mtSqr",float, doc= 'transverse mass squared'),
+      px = Var("px",float, doc= 'x coordinate of momentum vector'),
+      py = Var("py",float, doc= 'y coordinate of momentum vector'),
+      pz = Var("pz",float, doc= 'z coordinate of momentum vector'),
+      theta = Var("theta",float, doc= 'momentum polar angle'),
+      rapidity = Var("rapidity", float, doc='rapidity'),
+      y = Var("y", float, doc='rapidity'),
+      vx = Var("vx", float, doc='x coordinate of vertex position'),
+      vy = Var("vy", float, doc='y coordinate of vertex position'),
+      vz = Var("vz", float, doc='z coordinate of vertex position'),
+      vertexChi2 = Var("vertexChi2", float, doc='chi-squares'),
+      vertexNdof = Var("vertexNdof", float, doc='Number of degrees of freedom,  Meant to be Double32_t for soft-assignment fitters'),
+      vertexNormalizedChi2 = Var("vertexNormalizedChi2", float, doc='chi-squared divided by n.d.o.f.'),
+      pdgId = Var("pdgId", int, doc='PDG identifier'),
+      status = Var("status", int, doc='status word'),
+      longLived = Var("longLived", bool, doc='is long lived?'),
+      massConstraint = Var("massConstraint", bool, doc='do mass constraint?'),
+      hasMasterClone = Var("hasMasterClone", bool, doc='This only happens if the concrete Candidate type is ShallowCloneCandidate'),
+      hasMasterClonePtr = Var("hasMasterClonePtr", bool, doc='This only happens if the concrete Candidate type is ShallowClonePtrCandidate'),
+      isElectron = Var("isElectron", bool, doc=''),
+      isMuon = Var("isMuon", bool, doc=''),
+      isStandAloneMuon = Var("isStandAloneMuon", bool, doc=''),
+      isGlobalMuon = Var("isGlobalMuon", bool, doc=''),
+      isTrackerMuon = Var("isTrackerMuon", bool, doc=''),
+      isCaloMuon = Var("isCaloMuon", bool, doc=''),
+      isPhoton = Var("isPhoton", bool, doc=''),
+      isConvertedPhoton = Var("isConvertedPhoton", bool, doc=''),
+      isJet = Var("isJet", bool, doc=''),
```

Comment:
these variables will add dependency on reconstruction of other object types (e.g. reco electron, muon etc.). I suggest to not use them for this round and rely only on `pdgId` that is already stored above
```suggestion
```

Reply:
Ok, ty for the info. But for muon there seem to be different types of it -- is this info not needed neither?

Reply:
Muon faking tau is usually not a big issue at HLT level. Perhaps, it is relevant for mu+tau cross-trigger, but even there, simple cut-based selection is enough to reduce Z->mumu background. Therefore, for online tagging we should more focus on jets and electrons (with lower priority) faking taus.

Reply:
Ok, thanks a lot for your explanation!

Reply:
Another question (a more general one), why are we using variables at reco level? Shouldn'w we look at the RAW level data?

Reply:
At L3, HLT uses the same code base as RECO, so for our use-cases the definition of variables for taus and PF candidates will be the same (from the code perspective) for HLT and offline reco. But we will need to add also some HLT specific collections (e.g. pixel tracks, L1 seeds) that could be useful for online trainings.